backend/copr_backend/background_worker_build.py

```diff
@@ -740,6 +749,87 @@ def _add_pubkey(self):
         self.log.info("Added pubkey for user %s project %s into: %s",
                       user, project, pubkey_path)
 
+    def _setup_for_user_ssh(self):
+        """
+        Setup the builder for user SSH
+        https://github.com/fedora-copr/debate/tree/main/user-ssh-builders
+
+        If the builder setup for user SSH becomes more complicated than just
+        installing the public key, we might want to move the code to a script
+        within `copr-builder` and call it here or from `copr-rpmbuild`. There
+        is no requirement for it to be here.
+        """
+        if not self.job.allow_user_ssh:
+            return
+        self._deploy_user_ssh()
+        self._set_default_expiration()
+
+    def _deploy_user_ssh(self):
+        """
+        Deploy user public key to the builder, so that they can connect via SSH.
+        """
```

Comment:
This is a nice first step, if we make sure there's no shell injection (shlex.quote again).  Long term it would be better to have a more controlled environment for root commands.

For some time we are facing subscription-manager headaches (subscribing also requires root access, and takes too much time during builder startup). It would be nice to be able to subscribe "on demand".  Or wait for the async subscription-manager action "on demand".  It would be nice to have a new "root" hook in resalloc doing these per-task preparation actions so we could drop the root access from Backend entirely.